dask/array/linalg.py

```diff
@@ -829,22 +849,43 @@ def svd(a):
             "Input ndim: {}\n".format(a.shape, a.ndim)
         )
     if nb[0] > 1 and nb[1] > 1:
-        raise ValueError(
+        raise NotImplementedError(
             "Array must be chunked in one dimension only. "
             "This function (svd) only supports tall-and-skinny or short-and-fat "
             "matrices (see da.linalg.svd_compressed for SVD on fully chunked arrays).\n"
             "Input shape: {}\n"
             "Input numblocks: {}\n".format(a.shape, nb)
         )
 
-    # Tall-and-skinny case
-    if nb[0] >= nb[1]:
-        u, s, v = tsqr(a, compute_svd=True)
-        return u, s, v
-    # Short-and-fat case
+    # Single-chunk case
+    if nb[0] == nb[1] == 1:
+        m, n = a.shape
+        k = min(a.shape)
+        mu, ms, mv = np.linalg.svd(np.ones((1, 1), dtype=a.dtype))
```

Comment:
We might still have the issue of `a` being on a GPU here. IIUC, this is to determine the appropriate dtypes after the SVD for the `meta` later on. Hopefully this suffices

```suggestion
        mu, ms, mv = np.linalg.svd(a._meta)
```

That's a size-0 array, but at least for ndarrays np.linalg.svd seems to be OK with it.

Reply:
Ah interesting, that does clear all the tests.  Good to know. 

What's the concern with `a` being on a GPU?  Does anything in that code look like it would pull it back onto the CPU?

Reply:
Actually that doesn't work on Python 3.6 for windows apparently:

```
dask\array\linalg.py:864: in svd
    mu, ms, mv = np.linalg.svd(a._meta)
C:\Miniconda3\envs\test-environment\lib\site-packages\numpy\linalg\linalg.py:1542: in svd
    _assertNoEmpty2d(a)
```

Any objections to me changing it back?  I'm not sure how to work around that otherwise.

Reply:
Hey @TomAugspurger I reverted it just to give CI time to run in case you're OK with it as it was.  Let me know if there's something else that can be done to avoid any issues on a GPU (I don't understand what they are or I'd happily experiment with other solutions).

Reply:
Yeah, I'm fine with changing it back for now.

I'm not 100% sure what would happen if `a` were on the GPU. `u`, `s`, and `v` will be fine, since they're returning CuPy arrays. But I don't know what `from_delayed` does when it gets GPU memory and `meta` is a NumPy array rather than a CuPy array.

cc @jakirkham or @pentschev if you think there's any followup needed here.

Reply:
Ah I see, maybe something like this would make more sense? 

`np.linalg.svd(np.ones_like(a._meta, shape=(1,1)))`

Reply:
> Ah I see, maybe something like this would make more sense?
> 
> `np.linalg.svd(np.ones_like(a._meta, shape=(1,1)))`

Yeah, we should do that, otherwise `mu`, `ms`, `mv` will all be NumPy-arrays, and using them below, e.g. `meta=mu`, becomes completely irrelevant, as it will default anyway to a NumPy array.

Reply:
It should actually use `ones_like_safe` from https://github.com/dask/dask/blob/519b62fed6fa577a52b9273e11ccd9ba6d49644b/dask/array/utils.py#L314-L323, to ensure it still works on NumPy < 1.17 as well.

Reply:
Ok thanks @pentschev.  To clarify on this:

> using them below, e.g. meta=mu, becomes completely irrelevant, as it will default anyway to a NumPy array.

I still need the dtype on the mu, ms, mv arrays though.  I wanted to make sure that you're not implying that there's some way to do this without meta arrays since it's only the dtype I'm after (which would be great).

Setting the dtype for all the svd results as `'float32' if a.dtype == np.float32 else 'float64'` is the numpy.linalg.svd logic afaict, so I could just do that too and not worry about needing the template call.  There's always the chance something changes in numpy though so I presume the actual invocation is better.